src/snowflake/connector/auth.py

```diff
@@ -463,6 +472,25 @@ def read_temporary_credentials(self, host, user, session_parameters):
                 host, user, MFA_TOKEN
             )
 
+    @staticmethod
+    def get_token_from_private_key(user, account, privatekey_path, key_password):
+        encoded_password = key_password.encode() if key_password is not None else None
```

Comment:
Type hints could help with this. If you declare `def get_token_from_private_key(user, account, privatekey_path, key_password: str):` then that says that `key_password` should not be None ever.

Reply:
I added type hints for user, account, and privatekey_path, but it's okay if key_password is None so I didn't add the type hint here. This could happen if the user has an unencrypted private key file and so doesn't set the environment variable SNOWSQL_PRIVATE_KEY_PASSPHRASE, which will then be returned as None.

Reply:
The right type hint for a nullable string is `Optional[str]`.

Reply:
Added!